src/components/ColorPicker/components/ColorPickerContent/ColorPickerContentComponent.jsx

```diff
@@ -14,25 +15,64 @@ const ColorPickerContentComponent = ({
   noColorText,
   colorStyle,
   ColorIndicatorIcon,
+  SelectedIndicatorIcon,
   shouldRenderIndicatorWithoutBackground,
-  NoColorIcon
+  NoColorIcon,
+  isBlackListMode,
+  colorsList,
+  isMultiselect
 }) => {
   const onClearButton = useCallback(() => {
     onValueChange(null);
   }, [onValueChange]);
+
+  const onColorClicked = useCallback(
+    color => {
+      if (!isMultiselect) {
+        onValueChange(color);
+        return;
+      }
+      console.log(color);
+      const colors = [...value];
+      console.log(colors);
+      if (colors.includes(color)) {
+        const index = colors.indexOf(color);
+        console.log(index);
+        if (index > -1) {
+          colors.splice(index, 1);
+        }
+      } else {
+        colors.push(color);
+      }
+      console.log(colors);
+      onValueChange(colors);
+    },
+    [onValueChange]
+  );
+
+  let colorsToRender;
+  // If whitelist mode and not all the colors exists in the colors option - render all options
+  if (!isBlackListMode && _difference(colorsList, contentColors).length !== 0) {
+    colorsToRender = contentColors;
+  } else {
+    colorsToRender = isBlackListMode ? _difference(contentColors, colorsList) : colorsList;
+  }
   return (
     <div className={cx("color-picker-content--wrapper", className)}>
       <div className={cx("color-picker")}>
-        {contentColors.map(color => {
+        {colorsToRender.map(color => {
           return (
             <ColorPickerItemComponent
               key={color}
               color={color}
-              onValueChange={onValueChange}
+              onValueChange={onColorClicked}
               value={value}
               shouldRenderIndicatorWithoutBackground={ColorIndicatorIcon && shouldRenderIndicatorWithoutBackground}
               colorStyle={colorStyle}
               ColorIndicatorIcon={ColorIndicatorIcon}
+              SelectedIndicatorIcon={SelectedIndicatorIcon}
+              isSelected={isMultiselect ? value.includes(color) : value === color}
```

Comment:
is value always an array in isMultSelect?

Reply:
yes

Reply:
👍